lib/client.js

```diff
@@ -64,8 +146,12 @@ function Client (broker, conn, req) {
       that._parsingBatch = 0
       var buf = empty
       buf = client.conn.read(null)
-
-      if (buf) {
+      var { ipFamily, ipAddress, data } = protocolDecoder(buf)
```

Comment:
This needs to be done only for the frist packet and it should be put behind an option.

Reply:
I agree, could you tell me how to detect when the first packet arrived ?
Since _parsingBatch is already decremented, should i change the if (_parsingBatch <= 0) ? 
How do you see this option, like preConnect, Authenticate, etc ... broker hooks ? Or if broker.trustProxy is true ?